### PR TITLE
running smoke tests in low resolution

### LINKED ISSUE #1198: run UI tests with various resolutions
see #1185

### PR DESCRIPTION
**Pull request by individual-it**
## Description
ability to run UI tests in various resolutions
currently running full screen of selenium docker (1360x882) and 800x600
I've also added a tag for skipping tests that fail in the lower resolution
currently only smoke tests run in the lower resolution

## Related Issue

- Fixes #1198

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>

































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>

































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>

































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>





























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>





























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>




























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by individual-it** [comment]
Questions to discuss:
- what resolutions to run?
- what set of tests to run with each resolution (single suites, smoke tests, full set of tests)?
- how to name skip tags for failing tests?

**Comment by pmaier1** [comment]
> what resolutions to run?

These should give us orientation, I think:
- https://ux.stackexchange.com/questions/98770/responsive-design-most-common-resolution-steps
- https://responsivedesign.is/strategy/page-layout/defining-breakpoints/

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFolders</strong> failed. Please find the screenshots inside ...</summary><p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFavorites</strong> failed. Please find the screenshots inside ...</summary><p>










</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIUpload</strong> failed. Please find the screenshots inside ...</summary><p>














</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>


















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>






























</p></details>

**Comment by individual-it** [comment]
currently the breakpoints in phoenix are:
``` 640px; 960px; 1200px; 1600px;```

we could run tests in between all the breakpoints, but that would explode the CI time, so we need to decide what are the most important resolutions

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by DeepDiver1975** [comment]
> `640px; 960px; 1200px; 1600px;`

okay - so what about using one table and eon smartphone size?

- iPad 768x1024
- iPhone X 375x812

**Comment by individual-it** [comment]
change the tests to use those resolutions, but they are the screen resolutions, not the browser resolutions.
But maybe that bit of the heading does not make much difference

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>










</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

File: tests/acceptance/features/webUIFiles/hiddenFile.feature [hunk 1]
```diff
@@ -11,6 +11,7 @@ Feature: Hide file/folders
 
   @smokeTest
   @issue-900
+  @skipOnIphoneResolution @issue-1270
```
Comment (DeepDiver1975) [inline tests/acceptance/features/webUIFiles/hiddenFile.feature]: #1270 was fixed meanwhile

Comment (individual-it) [reply tests/acceptance/features/webUIFiles/hiddenFile.feature]: #1270 is about the create `+ NEW` button

**Comment by individual-it** [comment]
@LukasHirt @paurakhsharma @DeepDiver1975 please review again
I've added an 'aria-label' to the file-actions-burger-button so it can be found easier
also the tests now use this button if its available and click on the correct "action" button depending on the resolution
The addressing of it is a bit crazy, so if you have a better idea please tell me

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 2]
```diff
@@ -46,10 +46,24 @@ const assertContentOFLocalFileIs = function (fullPathOflocalFile, expectedConten
...
 
-Before(function (client) {
+Before(function () {
   createdFiles = []
+  if (typeof process.env.SCREEN_RESOLUTION !== 'undefined' && process.env.SCREEN_RESOLUTION.trim() !== '') {
+    let resolution = process.env.SCREEN_RESOLUTION.split('x')
+    resolution[0] = parseInt(resolution[0])
+    resolution[1] = parseInt(resolution[1])
+    if (resolution[0] > 1 && resolution[1] > 1) {
+      client.resizeWindow(parseInt(resolution[0]), parseInt(resolution[1]))
```
Comment (paurakhsharma) [inline tests/acceptance/stepDefinitions/generalContext.js]:
```suggestion
      client.resizeWindow(resolution[0], resolution[1])
```

`parseInt()` is redundant here

Comment (individual-it) [reply tests/acceptance/stepDefinitions/generalContext.js]: just to make sure there is no random rubish in the string. its a user input - never trust the user

Comment (individual-it) [reply tests/acceptance/stepDefinitions/generalContext.js]: you are right, there is a parse before that also, will change it

Comment (individual-it) [reply tests/acceptance/stepDefinitions/generalContext.js]: @paurakhsharma done

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Pull request merged**